### PR TITLE
Remove choices_as_values

### PR DESCRIPTION
**Pull request by seriy**
Remove choices_as_values to avoid deprecation in SF3.1 and greater

### CODE REVIEW

File: Form/Type/PhoneNumberType.php [hunk 1]
```diff
@@ -81,7 +81,11 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 $choiceType = 'Symfony\\Component\\Form\\Extension\\Core\\Type\\ChoiceType';
                 $textType = 'Symfony\\Component\\Form\\Extension\\Core\\Type\\TextType';
                 $countryOptions['choice_translation_domain'] = false;
-                $countryOptions['choices_as_values'] = true;
+
+                // To be removed when dependency on Symfony Form is bumped to 3.1.
+                if (\Symfony\Component\HttpKernel\Kernel::VERSION_ID < 30100) {
```
Comment (xabbuh) [inline Form/Type/PhoneNumberType.php]: Using the version from the HttpKernel is not a good idea as the installed versions of the Form and the HttpKernel components may differ.

Comment (seriy) [inline Form/Type/PhoneNumberType.php]: I agree, my mistake

Comment (thewilkybarkid) [inline Form/Type/PhoneNumberType.php]: Looks like the only possibility is checking if `TextType`doesn't implement `DataTransformerInterface`.

**Comment by thewilkybarkid** [comment]
Thanks, I tweaked it to avoid using reflection.

**Pull request merged**